symengine/dict.h

```diff
@@ -9,6 +9,8 @@
 #define SYMENGINE_DICT_H
 
 #include <gmpxx.h>
+#include <piranha/mp_integer.hpp>
+#include <piranha/hash_set.hpp>
```

Comment:
So this I would be careful with. I would only include Piranha from a .cpp file and keep it local. First, so that it can be turned off when Piranha is not installed, and second so that it doesn't slow down the compilation, which it would if you do it like this.